src/main.rs

```diff
@@ -52,15 +52,24 @@ fn main() -> Result<(), CairoRunError> {
     let cairo_run_config = cairo_run::CairoRunConfig {
         entrypoint: &args.entrypoint,
         trace_enabled,
-        print_output: args.print_output,
         layout: &args.layout,
         proof_mode: args.proof_mode,
     };
+
+    let file = std::fs::File::open(args.filename).map_err(|e| CairoRunError::Program(e.into()))?;
+
+    let mut buffer = std::io::BufWriter::new(std::io::stdout());
+
     let cairo_runner = match cairo_run::cairo_run(
-        &args.filename,
+        file,
         &cairo_run_config,
         args.secure_run,
         &mut hint_executor,
+        if args.print_output {
+            Some(&mut buffer)
```

Comment:
The whole output management is getting fucked by the `no_std`.
The current state of my work is that:
- we cannot use `io` stuff inside the lib without breaking compatibility
- you guys removed the `vm` from the return type of the `cairo_run` function, and the `vm` is a private field of the runner returned, so nobody can access the output content except by the "passing a buffer to the function way"
- not every user want to have the program output prefixed by `Program output:`, only your binary need it

As a result:
- the buffer passed to the `cairo_run` func should be no_std compatible, aka a growable Vec or String. The caller can then decide to write it the stdout or wherever he want
- the ``Program output:` prefix should disappear, it's the binary responsibility to prefix the output, not the lib one

A better way should be to entirely get rid from the `print_output` logic and make the program output available again for the library user by implementing a function `get_output` on the runner that would forward the call to it's private instance of the VirtualMachine

Reply:
> the "Program output:" prefix should disappear, it's the binary responsibility to prefix the output, not the lib one

This is actually the solution I had in mind. I thought I had posted it in a comment 🤔 
But since it's this PR that removes it from the lib, it should be this PR that adds it to the binary.